### PR TITLE
fix(helper): update vue 2 template and helper

### PR DESCRIPTION
**Pull request by eunjae-lee**
## What?

This PR updates the vue 2 template. Vue 2 helper + template used to have provider / inject pattern, and now that we've removed it, there is no such thing as `useFieldPlugin` in the vue 2 helper and the template. So what if we just remove vue 2 helper at all? We just need a little bit of snippet in the vue 2 template, and we're not actively supporting vue 2 anyway.

## Why?

JIRA: EXT-1958

### CODE REVIEW

**Comment by eunjae-lee** [comment]
Current dependencies on/for this PR:
* main
  * **PR #260** <a href="https://app.graphite.dev/github/pr/storyblok/field-plugin/260" target="_blank">Graphite</a> 
    * **PR #266** <a href="https://app.graphite.dev/github/pr/storyblok/field-plugin/266" target="_blank">Graphite</a> 
      * **PR #267** <a href="https://app.graphite.dev/github/pr/storyblok/field-plugin/267" target="_blank">Graphite</a> 
        * **PR #269** <a href="https://app.graphite.dev/github/pr/storyblok/field-plugin/269" target="_blank">Graphite</a> 
          * **PR #270** <a href="https://app.graphite.dev/github/pr/storyblok/field-plugin/270" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/storyblok/field-plugin/270?utm_source=stack-comment).

**Comment by demetriusfeijoo** [comment]
@eunjae-lee, I think removing all the vue2 helper would not be bad, IMO. 

However, how should we handle the [FieldPlugin file](https://github.com/storyblok/field-plugin/blob/main/packages/cli/templates/vue2/src/components/FieldPlugin.vue)?

I think we could remove this file since it seems not to be used anyway. what do you think?

**Comment by eunjae-lee** [comment]
@demetriusfeijoo if we get rid of vue 2 helper, then we should actually delete [this folder](https://github.com/storyblok/field-plugin/tree/main/packages/field-plugin/helpers/vue2), but I didn't yet because we haven't decided yet.

And actually that empty FieldPlugin.vue file is a placeholder for users. We provide an example inside the template, but users can delete the whole example and start from scratch with the FieldPlugin.vue file (which is bare minimum)

**Comment by demetriusfeijoo** [comment]
> And actually that empty FieldPlugin.vue file is a placeholder for users. We provide an example inside the template, but users can delete the whole example and start from scratch with the FieldPlugin.vue file (which is bare minimum)

Oh, I see @eunjae-lee.

I think once we stop using the `FieldPluginProvider` inside the template, we can remove also the helper, for sure. 
This way we avoid some developers using it and blocking us from removing it easily in the future?!

I mean, I was checking here, and after removing the `FieldPluginProvider` we would have anything else to offer from the helper, right? As we're not going to support this version, maybe the best way would be to remove the helper, indeed.

I was taking a better look at how the new approach would look and I started asking myself on how to make the logic below reusable between both [`FieldExample/index.vue`](https://github.com/storyblok/field-plugin/blob/main/packages/cli/templates/vue2/src/components/FieldPluginExample/index.vue) and [`FieldPlugin.vue`](https://github.com/storyblok/field-plugin/blob/main/packages/cli/templates/vue2/src/components/FieldPlugin.vue) files.

image

Do you have any idea here? As we don't have `hooks` here (for `useFieldPlugin()`), may we use `Vue mixins`?! 
What do you think @eunjae-lee?

**Comment by eunjae-lee** [comment]
> Do you have any idea here? As we don't have `hooks` here (for `useFieldPlugin()`), may we use `Vue mixins`?! What do you think @eunjae-lee?

@demetriusfeijoo hey, thanks for pointing this out. I forgot about this part. added https://github.com/storyblok/field-plugin/pull/270/commits/539f5d24732db7c92a87174d9ec9be9bb5f14d05

**Comment by demetriusfeijoo** [comment]
Hey @eunjae-lee, thanks for the adjustment. 🙌

What about getting rid of the helper? Do you have a better position on that?

File: scripts/copy-helpers.mjs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env zx
```
Comment (eunjae-lee) [inline scripts/copy-helpers.mjs:1]: I've also replaced `copy-helpers.sh` with `copy-helpers.mjs` which seems easier to maintain.

File: scripts/copy-helpers.mjs [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+import { $ } from 'zx'
+
+const helpers = ['react', 'vue3', 'vite']
+
+// eslint-disable-next-line functional/no-loop-statement
+for (const helper of helpers) {
+	await $`mkdir -p ./packages/field-plugin/dist/${helper}`
+	await $`cp -r ./packages/field-plugin/helpers/${helper}/dist/* ./packages/field-plugin/dist/${helper}`
```
Comment (demetriusfeijoo) [inline scripts/copy-helpers.mjs:10]:
```suggestion
  await $`mkdir -p ./packages/field-plugin/dist/${helper}`
  await $`cp -r ./packages/field-plugin/helpers/${helper}/dist/* ./packages/field-plugin/dist/${helper}`
```

Just found this improvement here, @eunjae-lee.

**Review by demetriusfeijoo — APPROVED** [review]
It looks good to me @eunjae-lee.

Thanks for you effort on this 🙌

**Comment by eunjae-lee** [comment]
### Merge Activity

* **Sep 14, 10:27 AM**: @eunjae-lee started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/storyblok/field-plugin/270).

**Pull request merged**